scripts/west_commands/runners/mdb.py

```diff
@@ -0,0 +1,127 @@
+        return 'mdb'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash', 'debug'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument('--jtag', default='digilent',
+                            help='choose the jtag interface')
+        parser.add_argument('--cores', default=1,
+                            help='choose the cores to debug')
+        parser.add_argument('--gui', default=True,
+                            help='enable gui debug')
+        parser.add_argument('--nsim_args', default='',
+                            help='if given, arguments for nsim simulator')
+        parser.add_argument('--dig-device', default='',
+                            help='''choose the digilent device when
+                             --jtag=digilent''')
+
+    @classmethod
+    def create(cls, cfg, args):
+        return MdbBinaryRunner(
+            cfg,
+            cores=args.cores,
+            jtag=args.jtag,
+            nsim_args=args.nsim_args,
+            gui=args.gui,
+            dig_device=args.dig_device
+            )
+
+    def do_run(self, command, **kwargs):
+        self.require(self.commander)
+
+        mdb_basic_options = ['-nooptions', '-nogoifmain',
+                            '-toggle=include_local_symbols=1']
+
+        # nsim
+        if self.nsim_args != '':
+            mdb_target = ['-nsim', '@'+self.nsim_args]
+        # hardware target
+        else:
+            if self.jtag == 'digilent':
+                mdb_target = ['-digilent', self.dig_device]
+            else:
+                mdb_target = ['']
+
+        if command == 'flash':
+            mdb_run = ['-run', '-cl']
+        else:
+            if self.gui:
+                mdb_run = ['-OKN']
+            else:
+                mdb_run = ['-cl']
+
+        if self.cores == '1':
+            mdb_cmd = ([self.commander] + mdb_basic_options + mdb_target +
+                       mdb_run + [self.elf_name])
+        elif self.cores == '2':
+            mdb_sub_cmd1 = ([self.commander] +
+                            ['-pset=1', '-psetname=core0'] +
+                             mdb_basic_options + mdb_target + [self.elf_name])
+            mdb_sub_cmd2 = ([self.commander] +
+                            ['-pset=2', '-psetname=core1', '-prop=download=2'] +
+                             mdb_basic_options + mdb_target + [self.elf_name])
+            mdb_cmd = ([self.commander] +
+                    ['-multifiles=core0,core1']
+                    + mdb_run)
+
+            self.check_call(mdb_sub_cmd1)
+            self.check_call(mdb_sub_cmd2)
+
+        elif self.cores == '4':
+            mdb_sub_cmd1 = ([self.commander] +
+                            ['-pset=1', '-psetname=core0'] +
+                             mdb_basic_options + mdb_target + [self.elf_name])
+            mdb_sub_cmd2 = ([self.commander] +
+                            ['-pset=2', '-psetname=core1', '-prop=download=2'] +
```

Comment:
Given pretty cryptic option name `-prop=download=2` I'd add a note right in the source saying what do we do here: inhibit loading of the same binary via other core's JTAG connections. And BTW that only makes sense in case of SMP-like system with use of only shared memory. Otherwise (like if some application code or data are located in either ICCM or DCCM which are private, i.e. available only on the core they belong to) we may get memory initialized in a pretty funny way :)

Reply:
I will some comments about it. I think SMP support is we need have now. For other cases, like AMP, or different core with different image, we can consider to add it  in the future.